module-1/README.md

```diff
@@ -63,55 +117,255 @@ export const environment = {
+  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    // The code that defines your stack goes here
+  }
+}
+```
+
+Following the change of filename and classname, you should now update the references in the `cdk.ts` file, as such.
+
+```typescript
+#!/usr/bin/env node
+import 'source-map-support/register';
+import cdk = require('@aws-cdk/cdk');
+import { Module1Stack } from '../lib/Module1Stack';
+
+const app = new cdk.App();
+new Module1Stack(app, 'Module1');
+```
+
+Now let's go through defining the S3 and CloudFront infrastructure.
+
+#### Define the S3 bucket
+
+First of all, we are going to define our S3 bucket and define the web index document as 'index.html'
+
+```typescript
+const projectName = "mythical-mysfits";
+const bucket = new s3.Bucket(this, "Bucket", {
+  bucketName: projectName + "-frontend-" + cdk.Aws.accountId,
+  websiteIndexDocument: "index.html"
+});
+```
+
+#### Restricted access to the S3 bucket
+
+Next we obtain the cloudfront origin access identity so that the s3 bucket may be restricted to it.  Then we define an IAM policy to restrict access to the S3 bucket so that only CloudFront distribution can access it.
+
+```typescript
+const origin = new cloudfront.CfnCloudFrontOriginAccessIdentity(this, "BucketOrigin", {
+  cloudFrontOriginAccessIdentityConfig: {
+    comment: projectName
+  }
+});
+
+const bucketPolicy = new iam.PolicyStatement()
+  .addPrincipal(new iam.CanonicalUserPrincipal(
+    origin.cloudFrontOriginAccessIdentityS3CanonicalUserId
+  ))
+  .addAction("s3:GetObject")
+  .addResource(bucket.bucketArn)
+  .addResource(bucket.bucketArn + "/*");
+bucket.addToResourcePolicy(bucketPolicy);
+```
+
+#### CloudFront Distribution
+
+Next, Write the definition for a new CloudFront web distribution.
+
+```typescript
+const cdn = new cloudfront.CloudFrontWebDistribution(this, "CloudFront", {
+  viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.AllowAll,
+  priceClass: cloudfront.PriceClass.PriceClassAll,
+  originConfigs: [
+    {
+      behaviors: [
+        {
+          isDefaultBehavior: true,
+          maxTtlSeconds: undefined,
+          allowedMethods:
+            cloudfront.CloudFrontAllowedMethods.GET_HEAD_OPTIONS
+        }
+      ],
+      originPath: `/web`,
+      s3OriginSource: {
+        s3BucketSource: bucket,
+        originAccessIdentityId: origin.cloudFrontOriginAccessIdentityId
+      }
+    }
+  ]
+});
+```
+
+#### CloudFormation Outputs
+
+Last of all, we want to define a cloudformation output for the domain name assigned to our CloudFront distribution.
+
+```typescript
+new cdk.CfnOutput(this, "CloudFrontURL", {
+  description: "The CloudFront distribution URL",
+  value: "https://" + cdn.domainName
+});
+```
+
+With that, we have completed writing the components of our module 1 stack.  
+
+### View the generated CloudFormation Scripts
+
+From within the `cdk` folder
+
+1. npm run build
+2. cdk synth
+
```

Comment:
Probably need to setup AWS default credentials + region first

Reply:
Did think this, but didn't go there because it wasnt covered in the original workshop.  Will reconsider. https://github.com/DavidChristiansen/aws-modern-application-workshop/issues/30